tests/test_on_data/test_metadata/test_neuralynx_metadata.py

```diff
@@ -13,56 +10,88 @@
 
 
 def test_neuralynx_cheetah_v574_metadata():
+    import neo
+    import distutils.version as version
+
     folder_path = NLX_PATH / "Cheetah_v5.7.4" / "original_data"
-    assert get_metadata(str(folder_path)) == dict(
-        session_start_time=datetime.datetime(2017, 2, 16, 17, 56, 4), session_id="d8ba8eef-8d11-4cdc-86dc-05f50d4ba13d"
-    )
+    metadata = NeuralynxRecordingInterface(folder_path).get_metadata()
+    file_metadata = metadata["NWBFile"]
+
+    assert file_metadata["session_start_time"] == datetime.datetime(2017, 2, 16, 17, 56, 4)
+    # the session id can only be extracted from the file headers exposed for neo > 0.11.0
+    if version.LooseVersion(neo.__version__) > version.LooseVersion("0.11.0"):
+        assert file_metadata["session_id"] == "d8ba8eef-8d11-4cdc-86dc-05f50d4ba13d"
+
+    # Metadata extracted directly from file header (neo >= 0.11)
+    assert '"FileType": "NCS"' in file_metadata["notes"]
+    assert '"recording_closed": "2017-02-16 18:01:18"' in file_metadata["notes"]
+    assert '"ADMaxValue": "32767"' in file_metadata["notes"]
+    # the sampling rate and device is part of the file headers exposed for neo > 0.11.0
```

Comment:
Is this > or >= ? 
I am asking because it seems that the latest dependency of spikeinterface is `neo>=0.11` . If this only works after neo > 0.11 then we should add this is a requirement. 

I also thing that the if statement is probably not necessary in the testing files as they have a a explicit environment in which they are meant to be run, maybe I am missing something, what do you think @CodyCBakerPhD ?

Reply:
The latest version of neo is 0.11.0. However, in the current dev version more metadata will be exposed, and this is what I am testing already here. I wasn't sure if you run the tests with multiple neo versions (or maybe some users run different versions), so I wanted to make the test as robust as possible right now. In the future if you ever upgrade your package dependency to neo >=0.12 we can simplify the code here again.

Reply:
> I wasn't sure if you run the tests with multiple neo versions (or maybe some users run different versions), so I wanted to make the test as robust as possible right now.

We do not, but we can (and should) add a dev gallery of tests and can (and should) include `neo` in that matrix. I'll see if I can get that setup today.


> In the future if you ever upgrade your package dependency to neo >=0.12 we can simplify the code here again.

Currently we do not pin `neo` on the `neuroconv` level, instead we rely on `spikeinterface` (atalystneuro/neuroconv/blob/main/src/neuroconv/datainterfaces/ecephys/requirements.txt), which in turn relies on latest patch version of `0.11` (https://github.com/SpikeInterface/spikeinterface/blob/master/requirements.txt#L2). Though if `spikeinterface does not update that after the next major `neo` release, we probably would here just to make reliance on this metadata stuff more reliable for our purposes.

Reply:
I added more specific comments and TODO for this.